src/main/webapp/3Dscript.js

```diff
@@ -45,144 +49,285 @@ function initThreeJs() {
 
 
 /**
- * This function fetchs pose data from the RunInfo servlet,
- * it is an asynchronous call requiring addMap() to be 
+ * Fetchs pose data from the RunInfo servlet,
+ * it is an asynchronous call requiring addMap() to be
  * called after the data is fully loaded.
  */
 function fetchData() {
-    const queryString = window.location.search;
-    const urlParams = new URLSearchParams(queryString);
-    const id = urlParams.get('id');
-    const type = urlParams.get('dataType');
-    fetch('/getrun?id=' + id + '&dataType=' + type)
-    .then(response => response.json())
-    .then(data => pose = data)
-    .then(() => addMap());
+  const queryString = window.location.search;
+  const urlParams = new URLSearchParams(queryString);
+  const id = urlParams.get('id');
+  const type = urlParams.get('dataType');
+  fetch('/getrun?id=' + id + '&dataType=' + type)
+      .then((response) => response.json())
+      .then((data) => pose = data)
+      .then(() => addMap());
+}
+
+/**
+ * Finds the median value of the latitude, longitude,
+ * and altitude. The median is used as a reference point
+ * for the East-North-Up (ENU) conversion and as the center
+ * point for the 2D map.
+ * @return {array} An array containing median lat, lng, and alt
+ *   coordinates.
+ */
+function findMedians() {
+  const arrays = {lat: [], lng: [], alt: []};
+  for (const point of pose) {
+    arrays.lat.push(point.lat);
+    arrays.lng.push(point.lng);
+    arrays.alt.push(point.alt);
+  }
+  const sorted = {
+    lat: [...arrays.lat].sort((a, b) => a - b),
+    lng: [...arrays.lng].sort((a, b) => a - b),
+    alt: [...arrays.alt].sort((a, b) => a - b)};
+  const mid = {
+    lat: sortedLat[Math.floor(sorted.lat.length / 2)],
+    lng: sortedLng[Math.floor(sorted.lng.length / 2)],
+    alt: sortedAlt[Math.floor(sorted.alt.length / 2)]};
+  return [mid.lat, mid.lng, mid.alt];
 }
 
 /**
  * Creates the 2D terrain and points a light at it.
- * It also calls two other functions so it can wait on the fetch 
- * request and significantly cut down rendering time with 
+ * It also calls two other functions so it can wait on the fetch
+ * request and significantly cut down rendering time with
  * geometry instancing.
 */
 function addMap() {
+  const [medianLat, medianLng, medianAlt] = findMedians();
+  median = {lat: medianLat, lng: medianLng, alt: medianAlt};
```

Comment:
Maybe you can define a Point class to make it easier to pass around this type of data?